insta2/src/components/CommentForm.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react'; 
+// import CommentSection from './CommentSection/CommentSection';
+
+class CommentForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+        comments: props.comments,
+        comment: ''
+        }
+    }
+
+    handleChange = event => {
+        this.setState({ [event.target.name]: event.target.value })
+    }
+
+    render() {
+        return (
+            <div>
+                <form onSubmit={this.props.addNewComment(this.props.postId, this.state.newComment)}>
+                    <input  
```

Comment:
when you need to create an event listener that will take in parameters, you need to make it an anonymous arrow function or it will fire immediately. this is the line of code that was causing the error in your comments map because it was adding the postId as a new comment key value on your state instead of making it an array, console log you state in your CommentForm.js render to see the keys that were created. the addNewComment method is trying to spread the state.comments, but a single integer is not iterable